src/Models/HydrostaticFreeSurfaceModels/SplitExplicitFreeSurfaces/split_explicit_free_surface.jl

```diff
@@ -250,12 +248,13 @@ end
 
 @inline function weights_from_substeps(FT, substeps, averaging_kernel)
 
-    τᶠ = range(FT(0), FT(2), length = substeps+1)
+    # To divide the range (t, t + 2dt):
+    Nτ = 2substeps + 1
```

Comment:
@simone-silvestri how can we determine the total substeps from the "user substeps" (ie the number of substeps that divide a time-step?

Do we need to introduce an interface where new averaging kernels are defined by:

1. The weights
2. The total number of substeps, as a function of the substeps that divide up a time step

?

We must be computing number 2 at some point (perhaps offline, or it is in a paper somewhere). We just need to elevate this information into this PR.